TicTacToe.js

```diff
@@ -1,310 +1,336 @@
+		return document.getElementById(id).innerHTML;
+	};
+
+	var game = function(){
+		//TODO create a board object that holds functions like update board, and create board.
+		var board = createBoard(); // init TTT board
+		var plr1 = new player("1"); // player 1 is 'X'
+		var plr2 = new playerRand("2"); // player 2 is 'O'
+		var isPlr1 = choseRand(); // choose start player
+		var nextMove;
+		var gameOver = -1;
+
+		//console.log(printBoard(board));
+
+		for(var i = 1; i <= 9; i++) { //there are only 9 tiles.
+			if(isPlr1) {
+				nextMove = plr1.move(board);
 			}else{
-			    isEmpty = false;
+				nextMove = plr2.move(board);
 			}
 
-			// reset column and row counters
-			if (loopCounter%3 === 0) {
-				columnTotal = 0;
-				rowTotal = 0;
+			if(isValidMove(nextMove, board)) {
+				//Player 1 goes
+				if(isPlr1) {
+					board[nextMove[0]][nextMove[1]] = "X";
+					printMove();
+					gameOver = isGameWon(board, "X");
+					if (gameOver === 1 ) {
+						updateWinCounter(1,"plr1");
+						return;
+					}else if (gameOver === 0) {
+						updateWinCounter();
+						return;
+					}else{
+						//for good measure
+					}
+				//Player 2 goes
+				}else{
+					board[nextMove[0]][nextMove[1]] = "O";
+					printMove();
+					gameOver = isGameWon(board, "O");
+					if (gameOver === 1) {
+						updateWinCounter(2,"plr2");
+						return;
+					}else if (gameOver === 0) {
+						updateWinCounter();
+						return;
+					 }else{
+						 //for good measure
+					 }
+				}
+			}else{
+				if(isPlr1){
+					invalidMove(1);
+					break;
+				}else{
+					invalidMove(2);
+					break;
+				}
 			}
+
+			isPlr1 = !isPlr1;
+		}
+		return;
+
+		function printMove(){
+			document.getElementById("ans").innerHTML += "<hr>" + printBoardHTML(board);
 		}
-	}
-	//check if the board is full but there are no winners
-	if(isEmpty === false) {
-	    return TIE;
-	}
-	return NOTOVER;
-}
 
+		function updateWinCounter(player, id){
+			if (id){
+				setInnerHTMLById("ans", getInnerHTMLById("ans") + "<h2>PLAYER "+player+" WINS THE GAME!</h2>");
+				document.getElementById(id).innerHTML += "I";
+			} else {
+				setInnerHTMLById("ans", getInnerHTMLById("ans") + "<h2>NO ONE WINS. ITS A TIE!</h2>");
+				document.getElementById("tie").innerHTML += "I";
+			}
+		}
+		function invalidMove(player){
+			setInnerHTMLById("ans", getInnerHTMLById("ans") + "<hr>" + printBoardHTML(board, nextMove));
+			setInnerHTMLById("ans", getInnerHTMLById("ans") + "Invalid move from Player "+player+"<br><h2>OPPONENT WINS THE GAME BECAUSE OF FORFAIT</h2>");
+		}
+	}
 
-//Helper function
-//sets the inner HTML of a given id to text
-function setInnerHTMLById(id, text) {
-	document.getElementById(id).innerHTML = text;
-}
+	var game10 = function() {
+		for(var i = 0; i < 10; i++) {
+        	game();
+        }
+	}
 
-//helper function
-//gets the innerHTML of a given id
-function getInnerHTMLById(id) {
-	return document.getElementById(id).innerHTML;
-}
+	//make only game and game10 public
```

Comment:
Ok.. I don't really get it yet. I need to play around with it and one day it will click and I'll understand fully :smile: